### ISSUE TITLE
WebSocket hangs

### ISSUE THREAD

**Issue by robertroeser** [ReactiveX/RxNetty#480]
Hi,

Been doing some testing with WebSockets and it looks like after a number of frames are sent that RXNetty is hanging. I changed the WebSocketClient test supply a range from 1 to 100k and it only makes to around 3k and just stops and hangs.

Thanks,
Robert

**Comment by NiteshKant** [comment]
@robertroeser Are you flushing the writes?

**Comment by robertroeser** [comment]
Guess not - I'm try to re-create a problem in reactive socket  so I maybe this isn't in rxnetty...Thought I re-created the problem here.

**Comment by NiteshKant** [comment]
The [echo example](https://github.com/ReactiveX/RxNetty/blob/0.5.x/rxnetty-examples/src/main/java/io/reactivex/netty/examples/http/ws/echo/WebSocketEchoClient.java#L93) does not flush explicitly as it writes a finite stream (RxNetty flushes on completion of the stream). So, if you increase the number of frames it writes, it will mostly block because of the channel buffer being full and hence the channel not being writable. If you flush the writes, then the data is written on the underlying socket. The easiest way to flush is `writeAndFlushOnEach()`.

I could repro the issue with the echo example by increasing the range, and it goes away if you add `flushOnEach`. Is this what you were trying?

**Comment by NiteshKant** [comment]
@robertroeser Did I answer your question? Can we close this issue?

**Comment by robertroeser** [comment]
No- trying to narrow down the exact problem, but it seems to be related to websockets and backpressure. Trying to come up with a test to replicate the problem without reactive sockets.

**Comment by robertroeser** [comment]
@NiteshKant I was able to run a test with a different websocket server and client. The only time the error reproduces is when I use the RXNetty server - it occurs with both the Netty client and the JSR client over websockets.

**Comment by NiteshKant** [comment]
@robertroeser can you share the code?

**Comment by robertroeser** [comment]
run the ping/pong test - it slow down after a few 100k requests
https://github.com/ReactiveSocket/reactivesocket-websocket-rxnetty
